test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReportGenerator.cs

```diff
@@ -0,0 +1,304 @@
+            };
+
+            // Value: (completedStatusCode, DirectMethodCompletedTime - EdgeHubRestartedTime)
+            Dictionary<HttpStatusCode, List<TimeSpan>> completedStatusHistogram = new Dictionary<HttpStatusCode, List<TimeSpan>>();
+
+            bool hasSenderResult = await this.SenderTestResults.MoveNextAsync();
+            bool hasReceiverResult = await this.ReceiverTestResults.MoveNextAsync();
+
+            while (hasSenderResult && hasReceiverResult)
+            {
+                this.ValidateResult(
+                    this.SenderTestResults.Current,
+                    this.Metadata.SenderSource,
+                    TestOperationResultType.EdgeHubRestartDirectMethod.ToString());
+
+                this.ValidateResult(
+                    this.ReceiverTestResults.Current,
+                    this.Metadata.ReceiverSource,
+                    TestOperationResultType.DirectMethod.ToString());
+
+                // Both sender & receiver have their dm results
+                resultCount[nameof(this.SenderTestResults)]++;
+                resultCount[nameof(this.ReceiverTestResults)]++;
+
+                // Adjust seqeunce number from both source to be equal before doing any comparison
+                EdgeHubRestartDirectMethodResult senderResult = JsonConvert.DeserializeObject<EdgeHubRestartDirectMethodResult>(this.SenderTestResults.Current.Result);
+                DirectMethodTestResult receiverResult = JsonConvert.DeserializeObject<DirectMethodTestResult>(this.ReceiverTestResults.Current.Result);
+
+                long senderSeqNum = this.ConvertStringToLong(senderResult.SequenceNumber);
+                long receiverSeqNum = this.ConvertStringToLong(receiverResult.SequenceNumber);
+
+                if (receiverSeqNum > senderSeqNum)
+                {
+                    // Increment sender result to have the same seq as the receiver
+                    await this.IncrementSequenceNumberAsync(
+                        this.SenderTestResults,
+                        nameof(this.SenderTestResults),
+                        receiverSeqNum,
+                        resultCount,
+                        completedStatusHistogram);
+
+                    // Fail the test
+                    isPassing = false;
+                }
+
+                if (receiverSeqNum < senderSeqNum)
+                {
+                    // Increment receiver result to have the same seq as the sender
+                    await this.IncrementSequenceNumberAsync(
+                        this.ReceiverTestResults,
+                        nameof(this.ReceiverTestResults),
+                        senderSeqNum,
+                        resultCount,
+                        completedStatusHistogram);
+
+                    // Fail the test
+                    isPassing = false;
+                }
+
+                // Note: In the current verification, if the receiver obtained more result dm than
```

Comment:
I don't think you will have duplicate sequence number in direct method scenario, right?

Reply:
From the sender logic, this should never happen. But if it does, let's make sure we fail it.

Reply:
I am confused here. The aim of this test is to check EdgeHub startup time. How does duplicate methods have anything to do with that?